src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+      type = 'Point';
+      this._redliningFeatures.on('add', this.handleTextAdding);
+    }
+
+    const drawInteraction = new OlInteractionDraw({
+      source: this.state.redliningLayer.getSource(),
+      type: type,
+      geometryFunction: geometryFunction,
+      style: this.getRedliningStyleFunction,
+      freehandCondition: OlEventsCondition.never
+    });
+
+    map.addInteraction(drawInteraction);
+
+    this.setState({
+      interaction: [drawInteraction]
+    }, () => {
+      drawInteraction.setActive(pressed);
+    });
+  }
+
+  /**
+   * Creates a correctly configured OL select and/or modify and/or translate
+   * interaction(s) depending on given editType and adds this/these to the map.
+   *
+   * @method
+   */
+  createSelectOrModifyInteraction = () => {
+    const {
+      editType,
+      map
+    } = this.props;
+
+    this._selectInteraction = new OlInteractionSelect({
+      condition: OlEventsCondition.singleClick,
+      style: this.getRedliningStyleFunction,
+      hitTolerance: 5
+    });
+
+    if (editType === 'Delete') {
+      this._selectInteraction.on('select', this.onFeatureRemove);
+    } else if (editType === 'Copy') {
+      this._selectInteraction.on('select', this.onFeatureCopy);
+    }
+
+    let interactions = [this._selectInteraction];
+
+    if (editType === 'Edit') {
+      const edit = new OlInteractionModify({
+        features: this._selectInteraction.getFeatures(),
+        deleteCondition: OlEventsCondition.singleClick,
+        style: this.getRedliningStyleFunction,
+        pixelTolerance: 10
+      });
+
+      edit.on('modifystart', this.onModifyStart);
+
+      const translate = new OlInteractionTranslate({
+        features: this._selectInteraction.getFeatures()
+      });
+      interactions.push(edit, translate);
+    }
+
+    interactions.forEach(i => map.addInteraction(i));
+
+    map.on('pointermove', this.onPointerMove);
+
+    this.setState({
+      interaction: interactions
+    });
+  }
+
+  /**
+   * Listener for `select` event of OL select interaction in `Delete` mode.
+   * Removes selected feature from the vector source and map.
+   *
+   * @param {Event} evt Event containing selected feature to be removed.
+   *
+   * @method
+   */
+  onFeatureRemove = evt => {
+    const feat = evt.selected[0];
+    this._selectInteraction.getFeatures().remove(feat);
+    this.state.redliningLayer.getSource().removeFeature(feat);
+    this.props.map.renderSync();
+  }
+
+  /**
+   * Listener for `select` event of OL select interaction in `Copy` mode.
+   * Creates a clone of selected feature and calls utility method to move it
+   * beside the original to avoid overlapping.
+   *
+   * @param {Event} evt Event containing selected feature to be copied.
+   *
+   * @method
+   */
+  onFeatureCopy = evt => {
+    const feat = evt.selected[0];
+    const copy = feat.clone();
+    //eslint-disable-next-line
+    const doneFn = finalFeature => {this._redliningFeatures.push(finalFeature)};
```

Comment:
Absolutely minor: Missing colon behind `this._redliningFeatures.push(finalFeature)`.

Reply:
fixed in 08a4ce5